pkg/solana/chainwriter/ata_creation.go

```diff
@@ -129,40 +129,49 @@ func (s *SolanaChainWriterService) handleATACreation(ctx context.Context, create
 		return fmt.Errorf("error enqueuing transaction: %w", err)
 	}
 
-	err = s.waitForTxFinality(ctx, ataUUID)
+	// Wait for ATA transaction to be confirmed (common Unconfirmed maps to Confirmed on Solana)
+	err = s.waitForTxStatus(ctx, ataUUID, types.Unconfirmed)
 	if err != nil {
 		return fmt.Errorf("error waiting for ATA transaction finality: %w", err)
 	}
 	return nil
 }
 
-func (s *SolanaChainWriterService) waitForTxFinality(ctx context.Context, transactionID string) error {
-	waitCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+func (s *SolanaChainWriterService) waitForTxStatus(ctx context.Context, transactionID string, desiredStatus types.TransactionStatus) error {
+	waitCtx, cancel := context.WithTimeout(ctx, s.config.TxStatusTimeout.Duration())
 	defer cancel()
 
-	ticker := time.NewTicker(1 * time.Second)
-	defer ticker.Stop()
+	backoff := 1 * time.Second
+	maxBackoff := 8 * time.Second
```

Comment:
It still seems to me that this should be taken out of `SubmitTransaction`. libocr expects transmit calls to be quick, and `8s` can cause the whole round to expire and be aborted.

Reply:
Oh wow, I wasn't aware of this constraint. This seems like a major issue... If we move this out of SubmitTransaction, where would it go? Some solana-specific hook before submit transaction in the CCIP plugin? CC: @amit-momin